system/Database/BaseBuilder.php

```diff
@@ -390,13 +397,18 @@ public function selectSum($select = '', $alias = '')
 	 * @return BaseBuilder
 	 * @throws \CodeIgniter\Database\Exceptions\DatabaseException
 	 */
-	protected function maxMinAvgSum($select = '', $alias = '', $type = 'MAX')
+	protected function maxMinAvgSum(string $select = '', string $alias = '', string $type = 'MAX')
 	{
-		if (! is_string($select) || $select === '')
+		if ($select === '')
 		{
 			throw new DatabaseException('The query you submitted is not valid.');
 		}
 
+		if (strpos($select, ',') !== false)
+		{
+			throw new DatabaseException('Only single column allowed.');
```

Comment:
Please add this as a new static method in the DatabaseException like the others. Also the message should be localized and stored in a lang file.